Missing POSIX/C++ headers that some libstdc++ setups pull in transitively.

env.cc: strcasecmp needs <strings.h>.
net.cc: std::map needs <map>.

--- a/src/include/debug.h
+++ b/src/include/debug.h
@@ -12,6 +12,9 @@
 #include "nccl_common.h"
 #include <stdio.h>
 #include <thread>
+#ifdef NCCL_OS_LINUX
+#include <strings.h>
+#endif
 #include "compiler.h"
 
 // Conform to pthread and NVTX standard
--- a/src/transport/net.cc
+++ b/src/transport/net.cc
@@ -22,6 +22,8 @@
 #include "compiler.h"
 #include <atomic>
 #include <assert.h>
+#include <map>
+#include <string>
 #include "graph.h"
 #include "graph/topo.h"
 #include "nccl_net.h"
